demisto_sdk/commands/content_graph/parsers/pack.py

```diff
@@ -0,0 +1,174 @@
+import logging
+from pathlib import Path
+from typing import Any, Dict, Iterator, List, Optional
+
+from demisto_sdk.commands.common.constants import MarketplaceVersions
+from demisto_sdk.commands.common.tools import get_json
+from demisto_sdk.commands.content_graph.common import (
+    ContentType,
+    PACK_METADATA_FILENAME,
+    PACK_CONTRIBUTORS_FILENAME,
+    Relationships
+)
+from demisto_sdk.commands.content_graph.parsers.base_content import BaseContentParser
+from demisto_sdk.commands.content_graph.parsers.content_item import ContentItemParser, NotAContentItem
+
+
+logger = logging.getLogger('demisto-sdk')
+
+
+class ContentItemsList(list):
+    """ An extension for list - a list of a specific content type.
+
+    Attributes:
+        content_type (ContentTypes): The content types allowed to be included in this list.
+    """
+    def __init__(self, content_type: ContentType):
+        self.content_type: ContentType = content_type
+        super().__init__()
+
+    def append_conditionally(self, content_item: ContentItemParser) -> bool:
```

Comment:
Why create a new method? Why not override the parent/super's `append` method?

Reply:
because it's a different signature - returns bool and not None

Reply:
Both methods can return a bool.